_posts/2017-11-02-totw-107.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "Tip of the Week #107: Reference Lifetime Extension"
+layout: tips
+sidenav: side-nav-tips.html
+published: draft
+permalink: tips/107
+type: markdown
+---
+
+# Tip of the Week #107: Reference Lifetime Extension
+
+Originally posted as totw/107 on 2015-12-10
+
+By Titus Winters (titus@google.com)
+
+There have been some reports of confusion about references and lifetimes after
+abseil.io/tips/101, so in this Tip we'll dive more into the question, "When does
+reference lifetime extension apply?"
+
+```c++
+string Foo::GetName();
+
+const string& name = obj.GetName();  // Is this safe/legal?
+```
+
+In short lifetimes of temporaries (referents) will be extended *if and only if*
+
+*   A local `const T&` (or `T&&`, although Google style generally ignores that)
```

Comment:
Reword?

(or `T&&`, although use of direct rvalue references is discouraged)

?

Reply:
I'd rather make it clear that it's only due to our style guidelines, rather than a broader discouragement.